### PR TITLE
adjust hysteresis to avoid initial over-deposit incentive

### PR DESCRIPTION
**Pull request by djrtwo**
Address #1609 

Make hysteresis calculations configurable.
Change default increment/decrement thresholds to x-0.25 and x+1.25

### CODE REVIEW

**Comment by protolambda** [comment]
From chat earlier, we need to consider that if we do not offset the actual effective balance, we:
- increase allowance for mistakes by 0.25
- increase punishment by requiring another 0.25 extra

```
Previously:
Downwards hysteresis width: 32-32 = 0
Upwards hysteresis width: 32.5-31 = 1.5
Now:
Downwards hysteresis width: 32-31.75 = 0.25
Upwards hysteresis width: 32.75 - 31 = 1.75
```

I think the system is closer to original intention if we change it to:
- A small margin of e.g. 0.05 for initially offline validators to not immediately lose 1 effective balance increment because of some small inactivity.
- The old 1.5 upwards width.

**Comment by mcdee** [comment]
What is the rationale behind x+0.75 rather than x+0.25?  A +/-0.25 would retain the current gap of requiring 0.5 Ether in rewards (or additional deposit) prior to an uptick in effective balance, and solves the issue of newly on-boarded deposits not suffering an immediate decrease in effective balance if they miss their first attestation.

**Comment by vbuterin** [comment]
Yeah, it should be -0.25 and +1.25, not +1.75.

> Downwards hysteresis width: 32-31.75 = 0.25
> Upwards hysteresis width: 32.75 - 31 = 1.75

The error here is that as implemeneted the "downwards hysteresis width" (wrong term arguably) is actually _negative_ 0.25, and not 0.25. So preserving the 1.5 width would require an upwards threshold of 1.25.

**Comment by protolambda** [comment]
Yeah the term could have been better. I care more about the hysteresis numbers than the names though, the leap needed to get back to 32 is affected significantly.

**Comment by djrtwo** [comment]
modded to be 0.25/1.25

Ready for review and merge @protolambda

**Comment by protolambda** [comment]
With experimentation and review these numbers may change again, I am thinking we could just make it configurable, and avoid further spec changes.

**Comment by djrtwo** [comment]
okay, now configurable @protolambda

File: configs/mainnet.yaml [hunk 1]
```diff
@@ -21,6 +21,12 @@ SHUFFLE_ROUND_COUNT: 90
 MIN_GENESIS_ACTIVE_VALIDATOR_COUNT: 16384
 # Jan 3, 2020
 MIN_GENESIS_TIME: 1578009600
+# 4
+HYSTERESIS_QUOTIENT: 4
+# 1 (minus 0.25)
+HYSTERESIS_DOWNWARD_MULTIPLIER: 1
+# 5 (plus 1.25)
+HYSTERESIS_UPWARD_MULTIPLIER: 5
```
Comment (protolambda) [inline configs/mainnet.yaml]: So previous behavior is, for context:
```
HYSTERESIS_QUOTIENT: 2
HYSTERESIS_DOWNWARD_MULTIPLIER: 0
HYSTERESIS_UPWARD_MULTIPLIER: 3
```

**Comment by Sintayew4** [comment]
#1627

**Pull request merged**